src/main/java/seedu/taassist/logic/commands/result/CommandResult.java

```diff
@@ -0,0 +1,126 @@
+package seedu.taassist.logic.commands.result;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+import seedu.taassist.logic.commands.actions.StorageAction;
+import seedu.taassist.logic.commands.actions.UiAction;
+import seedu.taassist.logic.commands.exceptions.CommandException;
+import seedu.taassist.logic.commands.exceptions.StorageActionException;
+import seedu.taassist.storage.Storage;
+
+/**
+ * Represents the result of a command execution.
+ */
+public class CommandResult {
+
+    public static final UiAction NO_UI_ACTION = UiAction.UI_NO_ACTION;
+    public static final StorageAction NO_STORAGE_ACTION = new NoStorageAction();
+    public static final String MESSAGE_NO_STORAGE_ACTION = "There are no storage actions to perform.";
+
+    private final String feedbackToUser;
+    private final UiAction uiAction;
+    private final StorageAction storageAction;
+
+    /**
+     * Constructs a {@code CommandResult} with the specified {@code feedbackToUser}.
+     */
+    public CommandResult(String feedbackToUser) {
+        requireNonNull(feedbackToUser);
+        this.feedbackToUser = feedbackToUser;
+        uiAction = NO_UI_ACTION;
+        storageAction = NO_STORAGE_ACTION;
+    }
+
+    /**
+     * Constructs a {@code CommandResult} with the specified {@code feedbackToUser} and {@code UiAction}.
+     */
+    public CommandResult(String feedbackToUser, UiAction uiAction) {
+        requireNonNull(feedbackToUser);
+        requireNonNull(uiAction);
+        this.feedbackToUser = feedbackToUser;
+        this.uiAction = uiAction;
+        storageAction = NO_STORAGE_ACTION;
```

Comment:
```suggestion
        this(feedbackToUser); 
        requireNonNull(uiAction); 
        this.uiAction = uiAction;
```
Similarly the constructor bellow.

Reply:
`this(feedbackToUser)` seems like invalid java syntax